src/config.rs

```diff
@@ -2460,6 +2489,29 @@ impl TiKvConfig {
         self.backup.validate()?;
         self.pessimistic_txn.validate()?;
         self.gc.validate()?;
+
+        let default_memory_usage_limit = Self::default_memory_usage_limit();
+        if self.memory_usage_limit.0 == 0 {
+            self.memory_usage_limit = default_memory_usage_limit;
+            return Ok(());
+        }
+        match self.memory_usage_limit.0.cmp(&default_memory_usage_limit.0) {
+            cmp::Ordering::Less => {
```

Comment:
Should we also limit the minimum memory to avoid setting the wrong value?

Reply:
It's a good idea but we can't determine the minimum value easily.